src/system_unix.cpp

```diff
@@ -0,0 +1,214 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2020 - 2023 Pionix GmbH and Contributors to EVerest
+
+#include "system_unix.hpp"
+
+#include <cassert>
+#include <stdexcept>
+
+#include <fcntl.h>
+#include <grp.h>
+#include <linux/securebits.h>
+#include <pwd.h>
+#include <signal.h>
+#include <sys/capability.h>
+#include <sys/prctl.h>
+#include <unistd.h>
+
+#include <fmt/core.h>
+
+namespace Everest::system {
+
+const auto PARENT_DIED_SIGNAL = SIGTERM;
+
+struct GetPasswdEntryResult {
+    explicit GetPasswdEntryResult(const std::string& error_) : error(error_) {
+    }
+
+    GetPasswdEntryResult(uid_t uid_, gid_t gid_, const std::vector<gid_t>& groups_) :
+        uid(uid_), gid(gid_), groups(groups_) {
+    }
+
+    std::string error;
+    uid_t uid;
+    gid_t gid;
+    std::vector<gid_t> groups;
+
+    operator bool() const {
+        return this->error.empty();
+    }
+};
+
+static GetPasswdEntryResult get_passwd_entry(const std::string& user_name) {
+    // Assuming that a user does not have more than 50 groups
+    constexpr int max_supplementary_groups = 50;
```

Comment:
That's probably not relevant "in the real world" but the real limit is certainly much higher :D

Reply:
agreed, but no idea what a good limit is here